### PR TITLE
[SPARK-13770] [Documentation][ML] Document the ML feature Interaction

### PR DESCRIPTION
**Pull request by chie8842**
I created Scala and Java example and added documentation.

### CODE REVIEW

File: docs/ml-features.md [hunk 1]
```diff
@@ -729,6 +729,28 @@ for more details on the API.
 </div>
 </div>
 
+## Interaction
+
+`Implements` is a `Transformer` which implements interaction transform.
```
Comment (srowen) [inline docs/ml-features.md]: "Interaction" is a Transformer ...

I think this examples is valuable but it actually says nothing about what the transformer does. I know it outputs the various interaction features but we should define and give an example here and/or in the comments of the examples.

Comment (hayashidac) [inline docs/ml-features.md]: Thank you for your comment. I added example and changed description.

**Comment by srowen** [comment]
Ping @hayashidac I'd be happy to work with you to get a more complete example into the docs for this operation. It could use them, yes.

**Comment by hayashidac** [comment]
@srowen sorry, I have some other task and will work on this tomorrow.

**Comment by hayashidac** [comment]
I changed input column format. Please check again.

**Comment by srowen** [comment]
Yeah I see what you've done, but the example still doesn't suggest what the output is. I think it's OK if the example has to be run to show the output, but, I think the documentation needs to exhibit a more representative input and output so that people understand what it does without having to run the example.

**Comment by hayashidac** [comment]
I reflected the contents of the indications. Please confirm it.

File: docs/ml-features.md [hunk 2]
```diff
@@ -729,6 +729,58 @@ for more details on the API.
 </div>
 </div>
 
+## Interaction
+
+`Interaction` is a `Transformer` which takes a vector/double columns, and generate a single vector column that contains multiplication results of all combination of each vector/double values.
```
Comment (srowen) [inline docs/ml-features.md]: Two small changes: "... which takes vector or double-valued columns, and generates a ..." 
"... contains the product of all combinations of one value from each input column"

File: docs/ml-features.md [hunk 3]
```diff
@@ -729,6 +729,58 @@ for more details on the API.
...
+
+For example, if you have two vector type columns each of which contains three double type values as input columns, then you'll get a vector with 9 double type values as the output column.
```
Comment (srowen) [inline docs/ml-features.md]: Suggest: "... each of which has 3 dimensions ..."  "... then you'll get a 9-dimensional vector ..."

**Review by srowen — CHANGES_REQUESTED** [review]
Looks good with a few edits to the text.

File: docs/ml-features.md (hunk 3 above)
Comment (hayashidac) [inline docs/ml-features.md]: Thank you for your check. I fixed the document for your comment.

**Review by srowen — APPROVED** [review]

**Comment by srowen** [comment]
Merged to master/2.1

**Pull request closed without merging**